relayer/src/supervisor/scan.rs

```diff
@@ -0,0 +1,749 @@
+        };
+
+        for chain in self.config.chains.clone() {
+            scans.chains.push(self.scan_chain(&chain));
+        }
+
+        scans
+    }
+
+    pub fn scan_chain(&mut self, chain_config: &ChainConfig) -> Result<ChainScan, Error> {
+        let span = info_span!("scan.chain", chain = %chain_config.id, );
+        let _guard = span.enter();
+
+        info!("scanning chain...");
+
+        let chain = match self.registry.get_or_spawn(&chain_config.id) {
+            Ok(chain_handle) => chain_handle,
+            Err(e) => {
+                error!(
+                    "aborting scan, reason: failed to spawn chain runtime with error: {}",
+                    e
+                );
+
+                return Err(Error::spawn(e));
+            }
+        };
+
+        let mut scan = ChainScan::new(chain_config.id.clone());
+
+        match self.use_allow_list(chain_config) {
+            Some(spec) if self.scan_mode == ScanMode::Auto => {
+                info!("chain uses an allow list, skipping scan for fast startup");
+                info!("allowed ports/channels: {}", spec);
+
+                self.query_allowed_channels(&chain, spec, &mut scan)?;
+            }
+            _ => {
+                info!("scanning chain for all clients, connections and channels");
+                self.scan_all_clients(&chain, &mut scan)?;
+            }
+        };
+
+        Ok(scan)
+    }
+
+    pub fn query_allowed_channels(
+        &mut self,
+        chain: &Chain,
+        spec: &ChannelsSpec,
+        scan: &mut ChainScan,
+    ) -> Result<(), Error> {
+        info!("querying allowed channels...");
+
+        for (port_id, channel_id) in spec.iter() {
+            let result = scan_allowed_channel(self.registry, chain, port_id, channel_id);
+
+            match result {
+                Ok(ScannedChannel {
+                    channel,
+                    counterparty_channel,
+                    connection,
+                    counterparty_connection_state,
+                    client,
+                }) => {
+                    let client_scan = scan
+                        .clients
+                        .entry(client.client_id.clone())
+                        .or_insert_with(|| ClientScan::new(client));
+
+                    let connection_scan = client_scan
+                        .connections
+                        .entry(connection.connection_id.clone())
+                        .or_insert_with(|| {
+                            ConnectionScan::new(connection, counterparty_connection_state)
+                        });
+
+                    connection_scan
+                        .channels
+                        .entry(channel.channel_id.clone())
+                        .or_insert_with(|| ChannelScan::new(channel, counterparty_channel));
+                }
+                Err(e) => error!("failed to scan channel {}, reason: {}", channel_id, e),
```

Comment:
I think we may have misleading error message.

See the error below:

> 2022-01-19T09:49:32.979160Z  INFO ThreadId(01) scan.chain{chain=agoricdev-7}: scanning chain...
2022-01-19T09:49:32.979407Z  INFO ThreadId(01) scan.chain{chain=agoricdev-7}: chain uses an allow list, skipping scan for fast startup
2022-01-19T09:49:32.979656Z  INFO ThreadId(01) scan.chain{chain=agoricdev-7}: allowed ports/channels: transfer/channel-11
2022-01-19T09:49:32.979923Z  INFO ThreadId(01) scan.chain{chain=agoricdev-7}: querying allowed channels...
2022-01-19T09:49:32.980282Z  INFO ThreadId(01) scan.chain{chain=agoricdev-7}:scan.channel{port_id=transfer channel_id=channel-11}: querying channel transfer/channel-11...
2022-01-19T09:49:33.605622Z ERROR ThreadId(01) scan.chain{chain=agoricdev-7}: failed to scan channel channel-11, reason: could not retrieve the connection hop underlying port/channel transfer/channel-11 on chain 'agoricdev-7'

Specifically, the part about "could not retrieve the connection hop underlying port/channel transfer/channel-11 on chain 'agoricdev-7' is not really correct. This is because port/channel `transfer/channel-11` on chain 'agoricdev-7' does not exist.

As a side note, transfer/channel-11 on chain 'agoricdev-7' is configured to be allowed:

>[chains.packet_filter]
 policy = 'allow'
 list = [
   ['transfer', 'channel-11'],
 ]

The channel details are `cosmos-transfer/channel-11` (port is `cosmos-transfer`, not `transfer`):

```
$ hermes -c ~/.hermes/testnets.toml query channel end agoricdev-7 transfer channel-11
2022-01-19T09:56:55.281061Z DEBUG ThreadId(01) Options: QueryChannelEndCmd { chain_id: ChainId { id: "agoricdev-7", version: 7 }, port_id: PortId("transfer"), channel_id: ChannelId("channel-11"), height: None }
Error: port 'transfer' & channel 'channel-11' does not exist

$ hermes -c ~/.hermes/testnets.toml query channel end agoricdev-7 cosmos-transfer channel-11
2022-01-19T09:57:11.923361Z DEBUG ThreadId(01) Options: QueryChannelEndCmd { chain_id: ChainId { id: "agoricdev-7", version: 7 }, port_id: PortId("cosmos-transfer"), channel_id: ChannelId("channel-11"), height: None }
Success: ChannelEnd {
    state: Open,
    ordering: Unordered,
    remote: Counterparty {
        port_id: PortId(
            "transfer",
        ),
        channel_id: Some(
            ChannelId(
                "channel-211",
            ),
        ),
    },
    connection_hops: [
        ConnectionId(
            "connection-15",
        ),
    ],
    version: Version(
        "ics20-1",
    ),
}
```

Reply:
23924e99

Reply:
Resolved, thanks!!